### ISSUE TITLE
[Blue Burst] [Question(s)] Regarding team points, overall ItemPMT.prs usage of the server and shop data location.

### ISSUE THREAD

**Issue by Bluuzeus** [fuzziqersoftware/newserv#567]
I am playing around with the idea of a progression system that allows one to get specific rares via Team Points, idea being that after getting X number of rares a team could claim a rare of their choosing. However, modifying ItemPMT.prs on the /blueburst folder does not seem to change how many team points are given for each rare, changes to PMT do however show up in the sense that the UI reflect the changes - just that the server does not seem to mirror these changes.

I had a similar experience changing mag feed tables, during gameplay the tables seemed to be working but when returning to the lobby the mag's stats would be rebalanced to vanilla values based on what I fed them (honestly even though this is not what I wanted, I am glad that worked so well, gg whoever coded that). So is the PMT file more of a UI thing in general? Am I supposed to mirror changes to it elsewhere?

Also, as a bit of a sidenote. Where are the shop data located, I would also like to make some changes to the NPC stores. Mainly looking for the ones that use meseta as currency, but the other could be useful as well.

**Comment by nolrinale** [comment]
In addition to changing the PMT you also have to let the server know the real values to keep track of, so for that I believe you want to adjust the points costs in the `config.example.json` as well.

The Team System items are hardcoded into the client with very specific logic that accounts for the amount of members in a team for example the Commander Blade behavior so adding/removing rewards might require further modifications to the game client

**Comment by Bluuzeus** [comment]
Where in the config do I let it know to use the PMT changes? I have not seen this option, but also maybe we are talking about different things?

Adding / Removing items from the config has worked fine so far, adding too many does boot you out of the lobby (something around 80-90 items total) when you try to open the purchase menu at the lobby and having a lot of items does add some lag to opening that same menu.

The issue I am trying to resolve is regarding the amount of team points you get for turning in items. So for example, if I change shields and frames to give 10 points (normally 0) they still give 0, if I change the Team Points 500 item to give 50 points instead it still gives 500. On the menu you use to turn in rares to get team points the UI does reflect changes I made to the PMT, but the server does not actually reflect those points when I turn those items in.

In short
Buying Items with Team Points from the Lobby: Works Great
Exchanging an item to receive Team Points: Works Great BUT the Points received does not mirror changes on the PMT.

**Comment by nolrinale** [comment]
Everything related to TeamPoints handled by the server is located from this line in your config.
https://github.com/fuzziqersoftware/newserv/blob/c6490cb3fbe5ca49d846fba9ecfae93bfc2eef84/system/config.example.json#L875

I think fuzzi mentioned once that Team Point support is still being checked on as theres also a missing feature of a Team Point exchange multiplier that is not implemented yet either but I'm sure he can explain in more detail everything.

**Comment by Bluuzeus** [comment]
Ok yeah, this is where I've been adding the new items to purchase with team points, this like the rest of the config I've messed around with so far works great, with the caveat that you cant add too many items. Can even add specific Tech disk levels and item stacks for items that generally do not stack (while having the item stacking patch configured and enabled).

**Comment by fuzziqersoftware** [comment]
Which file are you editing? The server loads system/item-tables/ItemPMT-bb.prs for its internal usage, and sends system/blueburst/ItemPMT.prs to the client. I'm guessing you're modifying the second file and not copying it to the first. Probably I should make one of these a symlink to the other, so this kind of thing will be less confusing in the future.

Re: team point rewards on the client, only two of them are hardcoded to affect behavior on the client: team flag and dressing room. The others are all implemented server-side.

**Comment by Bluuzeus** [comment]
Perfect! Just tested this and yeah, this solved the points exchange and I can even add items that normally cannot be turned into team points and it also fixed the other issue I mentioned where changing the mag feed tables would lead to a missmatch of mag stats since the feeding stats the client would see and the server were different.

The only other open item is regarding shops. 

I'm looking for a way to edit the shops in game, mainly the meseta shops, but the other shops could be useful. Is this something I can edit in the server files? Or is this a client thing? I have not seen much in regards to tools that even do this in the few years I've messed around with BB, so figured I should ask here since you guys have so much experience with the workings of the game.

**Comment by nolrinale** [comment]
For shopping I think the easiest approach is to create a quest and put all your custom shop logic in there for the items you want to offer/ask for.

The shops were reverse engineered from the PSOGC version (SEGA just directly ported them over to BB without changes) you can check them out in the file
https://github.com/fuzziqersoftware/newserv/blob/master/src/ItemCreator.cc

It has logic regarding the lists, tekker, generating the items, and everything that may be interesting for you
I'm sure it's possible to change the behavior to what you expect but I think it wont be a trivial effort.

**Comment by Bluuzeus** [comment]
Wooooooow thanks a ton for sharing this, just looking over it is so insightful. There are questions I did not even know I had until I saw them explained on the comments there.

One change I would like to make to the shops is to allow the tool shop to sell max level techs even if they are very uncommon. I can see the code that seems to be associated with this behavior, could I just make a clone with Visual Studio make some changes and rebuild or is this already stepping into this wont be a trivial effort territory?

Thanks again for linking the file, I feel like the kid me playing this on GC would have its mind blown looking at something like this.

**Comment by nolrinale** [comment]
Nothing is stopping you from doing any changes you want, thats why fuzzi documented so many things in the codebase for whoever that controls C/C++ can modify and rebuild their own solution the way they want. 

You just modify the source files you want and build the solution following the instructions on the readme as usual

Consider closing this ticket if you think all your questions are answered so we can keep track of the proper bugs we find along the way.

**Comment by Bluuzeus** [comment]
Golden, thanks for the help and all the insight - closing this up! <3
